File: 0047-permutations-ii/0047-permutations-ii.py
class Solution:
    def permuteUnique(self, nums):
        def backtrack(start):
            if start == len(nums):
                ans.append(nums[:])
                return
            
            used = set()
            for i in range(start, len(nums)):
                if nums[i] in used:
                    continue 
                used.add(nums[i])
                nums[start], nums[i] = nums[i], nums[start]
                backtrack(start + 1)
                nums[start], nums[i] = nums[i], nums[start] 
        
        nums.sort()  
        ans = []
        backtrack(0)
        return ans

File: 0047-permutations-ii/0047-permutations-ii.cpp
class Solution {
public:
    vector<vector<int>> permuteUnique(vector<int>& nums) {
        vector<vector <int>>res;
        set<vector<int>>s;
        sort(nums.begin(),nums.end());
        do{
            s.insert(nums);

        }while(next_permutation(nums.begin(),nums.end()));
        for(auto i: s){
            res.push_back(i);
        }
        return res;
        
    }
};